src/Ship.cpp

```diff
@@ -361,6 +364,87 @@ void Ship::UpdateMass()
 	SetMass((m_stats.static_mass + GetPropulsion()->FuelTankMassLeft()) * 1000);
 }
 
+template <typename T>
+inline int sign(T num)
+{
+	return (num >= 0) - (num < 0);
+}
+
+vector3d Ship::CalcAtmosphericForce() const
+{
+	// Data from ship.
+	double m_topCrossSec = GetShipType()->topCrossSection;
+	double m_sideCrossSec = GetShipType()->sideCrossSection;
+	double m_frontCrossSec = GetShipType()->frontCrossSection;
+
+	// TODO: vary drag coefficient based on Reynolds number, specifically by
+	// atmospheric composition (viscosity) and airspeed (mach number).
+	double m_topDragCoeff = GetShipType()->topDragCoeff;
+	double m_sideDragCoeff = GetShipType()->sideDragCoeff;
+	double m_frontDragCoeff = GetShipType()->frontDragCoeff;
+
+	double m_shipLiftCoeff = GetShipType()->shipLiftCoefficient;
+
+	// By converting the velocity into local space, we can apply the drag individually to each component.
+	auto m_localVel = GetOrient().Inverse() * GetVelocity();
```

Comment:
IIRC, `auto m_localVel = GetOrient().Inverse() * GetVelocity();` is the same as
`auto m_localVel = GetVelocity() * GetOrient();` but without the needs of `Inverse`

Reply:
My matrix math admittedly isn't quite up to par, but `operator* (Vector<T>, matrix4x4)` does a *transpose* operation, not a matrix inversion. We're specifically converting a vector into local space (the Orient matrix is local-to-world), so the inverse of the world-space orientation matrix is what is required here.

Reply:
The rotation/orientation matrices are orthogonal. For an orthogonal matrix, the inverse and transpose are identical. Hence the Inverse() has the same results as Transpose() but is far slower.

Reply:
I replaced the line with what @mike-f1 suggested.